geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/AttributeOrdering.scala

```diff
@@ -0,0 +1,147 @@
+    }
+    ordering.asInstanceOf[Ordering[AnyRef]]
+  }
+
+  val StringOrdering   : Ordering[String]            = new NullOrdering(Ordering.String)
+  val IntOrdering      : Ordering[Integer]           = new NullOrdering(Ordering.ordered[Integer])
+  val LongOrdering     : Ordering[java.lang.Long]    = new NullOrdering(Ordering.ordered[java.lang.Long])
+  val FloatOrdering    : Ordering[java.lang.Float]   = new NullOrdering(Ordering.ordered[java.lang.Float])
+  val DoubleOrdering   : Ordering[java.lang.Double]  = new NullOrdering(Ordering.ordered[java.lang.Double])
+  val BooleanOrdering  : Ordering[java.lang.Boolean] = new NullOrdering(Ordering.ordered[java.lang.Boolean])
+  val DateOrdering     : Ordering[Date]              = new NullOrdering(Ordering.ordered[Date])
+  val UuidOrdering     : Ordering[UUID]              = new NullOrdering(Ordering.ordered[UUID])
+  val GeometryOrdering : Ordering[Geometry]          = new NullOrdering(new GeometryOrdering())
+  val BytesOrdering    : Ordering[Array[Byte]]       = new NullOrdering(new BytesOrdering())
+
+  val StringListOrdering   : Ordering[java.util.List[String]]            = ListOrdering(StringOrdering)
+  val IntListOrdering      : Ordering[java.util.List[Integer]]           = ListOrdering(IntOrdering)
+  val LongListOrdering     : Ordering[java.util.List[java.lang.Long]]    = ListOrdering(LongOrdering)
+  val FloatListOrdering    : Ordering[java.util.List[java.lang.Float]]   = ListOrdering(FloatOrdering)
+  val DoubleListOrdering   : Ordering[java.util.List[java.lang.Double]]  = ListOrdering(DoubleOrdering)
+  val BooleanListOrdering  : Ordering[java.util.List[java.lang.Boolean]] = ListOrdering(BooleanOrdering)
+  val DateListOrdering     : Ordering[java.util.List[Date]]              = ListOrdering(DateOrdering)
+  val UuidListOrdering     : Ordering[java.util.List[UUID]]              = ListOrdering(UuidOrdering)
+  val GeometryListOrdering : Ordering[java.util.List[Geometry]]          = ListOrdering(GeometryOrdering)
+  val BytesListOrdering    : Ordering[java.util.List[Array[Byte]]]       = ListOrdering(BytesOrdering)
+
+  private class GeometryOrdering extends Ordering[Geometry] {
+    override def compare(x: Geometry, y: Geometry): Int = x.compareTo(y)
+  }
+
+  private class BytesOrdering extends Ordering[Array[Byte]] {
+    override def compare(x: Array[Byte], y: Array[Byte]): Int = {
+      val len = math.min(x.length, y.length)
+      var i = 0
+      while (i < len) {
+        val res = java.lang.Byte.compare(x(i), y(i))
+        if (res != 0) {
+          return res
+        }
+        i += 1
+      }
+      Integer.compare(x.length, y.length)
+    }
+  }
+
+  private object ListOrdering {
+    def apply[T](delegate: Ordering[T]): Ordering[java.util.List[T]] = new NullOrdering(new ListOrdering(delegate))
+  }
+
+  private class ListOrdering[T](delegate: Ordering[T]) extends Ordering[java.util.List[T]] {
+    override def compare(x: java.util.List[T], y: java.util.List[T]): Int = {
```

Comment:
@jnh5y what do you think about this as an ordering for lists? compare elements up to `min(x.length, y.length)`, then take the smaller list. there doesn't seem to be any canonical list ordering in java that i could find.

Reply:
Let's try this by an analogy...  If a string is a List[Char], it sounds like your list ordering would function the same as the ordering for String?  (More or less?)

Seems reasonable to me!

Reply:
good point!